### PR TITLE
Quick links to members in markdown 

### PR DESCRIPTION
**Pull request by sksavant**
Feature #504

### CODE REVIEW

**Comment by sksavant** [comment]
WIP : Needs escaping regex match for member matches

File: lib/haml/filters/growstuff_markdown.rb [hunk 1]
```diff
@@ -3,6 +3,8 @@
 module Haml::Filters
   module GrowstuffMarkdown
     CROP_REGEX = /\[([^\[\]]+?)\]\(crop\)/
+    MEMBER_REGEX = /\[([^\[\]]+?)\]\(member\)/
+    MEMBER_AT_REGEX = /(\@[^\[ \n]+)/
```
Comment (pozorvlak) [inline lib/haml/filters/growstuff_markdown.rb]: Do you mean "an `@` followed by one or more non-whitespace characters"? I'd express that as `/@\S+/`. But actually, we're more restrictive about the characters allowed in usernames - we only allow [letters, numbers and underscores](https://github.com/Growstuff/growstuff/blob/dev/app/models/member.rb#L76). What you *really* want is `/@\w+/`.

File: lib/haml/filters/growstuff_markdown.rb [hunk 2]
```diff
@@ -20,6 +22,32 @@ def render(text)
         end
       end
 
+      # turn [jane](member) into [jane](http://growstuff.org/members/jane)
+      expanded = expanded.gsub(MEMBER_REGEX) do |m|
+        member_str = $1
+        # find crop case-insensitively
```
Comment (pozorvlak) [inline lib/haml/filters/growstuff_markdown.rb]: I think you mean "find member case-insensitively" :-)

File: lib/haml/filters/growstuff_markdown.rb [hunk 3]
```diff
@@ -20,6 +22,32 @@ def render(text)
...
+          "[#{member_str}](#{url})"
+        else
+          member_str
+        end
+      end
+
+      # turn @jane into [@jane](http://growstuff.org/members/jane)
+      expanded = expanded.gsub(MEMBER_AT_REGEX) do |m|
+        member_str = $1
+        # find crop case-insensitively
```
Comment (pozorvlak) [inline lib/haml/filters/growstuff_markdown.rb]: Again, s/crop/member.

File: spec/lib/haml/filters/growstuff_markdown_spec.rb [hunk 4]
```diff
@@ -60,4 +74,27 @@ def output_link(crop, name=nil)
...
+  end
+
+  it 'converts @ member links' do
+    @member = FactoryGirl.create(:member)
+    rendered = Haml::Filters::GrowstuffMarkdown.render("@#{@member.login_name}")
+    rendered.should match /#{output_member_link(@member, "@#{@member.login_name}")}/
+  end
+
+  it "doesn't convert nonexistent @ members" do
+    rendered = Haml::Filters::GrowstuffMarkdown.render("@not-a-member")
```
Comment (pozorvlak) [inline spec/lib/haml/filters/growstuff_markdown_spec.rb]: This isn't a legal member name. It's probably worth testing "invalidly-formatted name" and "valid but unused name" separately.

**Comment by pozorvlak** [comment]
Aha, I see you've already fixed most of the problems I noticed! This is what I get for opening a browser tab before the weekend and not looking at it until now. It looks great, apart from the `not-a-member` test using an invalid name, which won't be matched - I think the best thing is to add another test that uses a valid but unused name. Also, please add yourself to `CONTRIBUTORS.md`.

Welcome to the project!

**Comment by pozorvlak** [comment]
Wait, you're already in `CONTRIBUTORS.md`. Sorry! Just the test to fix, then.

**Comment by sksavant** [comment]
@pozorvlak: Added the test and merged in dev changes.

**Pull request merged**